.github/workflows/shark_model_suite.yml

```diff
@@ -0,0 +1,91 @@
+# Runs the shark model suite on iree wheels from build package workflow
+name: Validate Shark Model Suite
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_id:
+        description: 'Release id to publish'
+        required: true
+      package_version:
+        description: 'Version of the package'
+        required: true
+      build_run_id:
+        description: 'Run ID for the build_package.yml workflow that triggered this workflow'
+        required: true
+
+jobs:
+  build-validate:
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [a100, cpu, gpu, Linux, postsubmit, self-hosted, trust=basic, X64]
+        suite: [cpu,gpu,vulkan]
+        python-version: ["3.10"]
+
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Set up Python Version File ${{ matrix.python-version }}
+        run: |
+          # See https://github.com/actions/setup-python/issues/433
+          echo ${{ matrix.python-version }} >> $GITHUB_WORKSPACE/.python-version
+      
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: '${{ matrix.python-version }}'
+          #cache: 'pip'
+          #cache-dependency-path: |
+          #  **/requirements-importer.txt
+          #  **/requirements.txt
+      - name: Download SHARK
+        id: download_shark
+        uses: actions/checkout@v2
+        with:
+          repository: nod-ai/SHARK
+          path: SHARK
+          ref: 20220708.119
+      - name: Download packages
+        id: download_packages
+        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2
+        with:
+          github_token: ${{secrets.WRITE_ACCESS_TOKEN}}
+          workflow: build_package.yml
+          run_id: ${{ github.event.inputs.build_run_id }}
+      - name: Extract and display downloaded files
+        run: |
+          tar -xf artifact/iree-dist-${{ github.event.inputs.package_version }}-linux-x86_64.tar.xz
+          pwd
+          ls -R
+      - name: Validate CPU Models
+        if: matrix.suite == 'cpu'
+        run: |
+          cd $GITHUB_WORKSPACE
+          cd SHARK
+          PYTHON=python${{ matrix.python-version }} USE_IREE=1 IMPORTER=1 ./setup_venv.sh
+          source shark.venv/bin/activate
+          python -m pip install --force-reinstall -f file://$PWD/artifact/ iree-compiler iree-runtime iree-tools-tflite iree-tools-tf iree-tools-xla
```

Comment:
I believe we want to build these from source at the given commit instead of pip install them. @GMNGeoffrey, can you advise on how to do that?

Reply:
It depends on what we're doing. This pip install is coming from a previously-built release, so seems fine, assuming we're running this at the cadence of releases. If we want something that's attached to a specific commit and posts statuses there, then it would need to be built from source.

Reply:
I modeled the file after the validate_and_publish_release.yml, which, if I understand correctly, as a pre-step builds the repo into a package from source, which is then passed to this workflow as an input. If you want to post to commits as @GMNGeoffrey mentioned I can try and figure out how to get it to work like that